modules/light-clients/08-wasm/testing/wasm_endpoint.go

```diff
@@ -0,0 +1,41 @@
+package wasmtesting
+
+import (
+	"github.com/stretchr/testify/require"
+
+	types "github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+	clienttypes "github.com/cosmos/ibc-go/v7/modules/core/02-client/types"
+	ibctesting "github.com/cosmos/ibc-go/v7/testing"
+)
+
+type WasmEndpoint struct {
+	*ibctesting.Endpoint
+}
+
+var (
+	contractClientState    = []byte{1}
+	contractConsensusState = []byte{2}
+)
+
+// CreateClient creates an IBC client on the endpoint. It will update the
+// clientID for the endpoint if the message is successfully executed.
+// NOTE: a solo machine client will be created with an empty diversifier.
+func (endpoint *WasmEndpoint) CreateClient() (err error) {
```

Comment:
took the quick route. I think the testing pkg will need to define either an interface for the Endpoint type or an interface that the endpoint needs to construct messages (such as a QueryProof interface fn). I think this will be easier to do reason about after we start the testing pkg improvement issues, so for now, I thought it was easier to duplicate the code a bit by implementing the fn's that require wasm client specific logic

Reply:
opened #4807 and #4808 for the direction I think makes the most sense. For now, I think we should move forward with continuing to create this override type. Essentially the WasmEndpoint is a mock cometbft chain with a wasm client on it, and the endpoint represents that client and its connection/channel. 

Once we implement interfaces for the light client module to plug into the endpoint, we could implement that interface and remove this type